src/virt.c

```diff
@@ -715,9 +758,18 @@ refresh_lists (void)
             xpath_ctx = xmlXPathNewContext (xml_doc);
 
             /* Block devices. */
+	    char * bd_xmlpath=NULL;
+	    if( source == blockdevice_format ){
+		    bd_xmlpath="/domain/devices/disk/source[@dev]";
+	    }else{
+			if( dev == blockdevice_format ){
+				bd_xmlpath="/domain/devices/disk/target[@dev]";
+			}
+	    }
             xpath_obj = xmlXPathEval
-                ((xmlChar *) "/domain/devices/disk/target[@dev]",
-                 xpath_ctx);
+                  ((xmlChar *) bd_xmlpath,
```

Comment:
The original code has terrible newline placement here. Please improve this by putting it all on one line:

```c
xpath_obj = xmlXPathEval ((xmlChar *) bd_xmlpath, xpath_ctx);
```